lib/streamlit/DeltaGenerator.py

```diff
@@ -1777,6 +1801,140 @@ def text_input(self, element, label, value=""):
         current_value = ui_value if ui_value is not None else value
         return str(current_value)
 
+    @_with_element
+    def number_input(
+        self,
+        element,
+        label,
+        value=NoValue(),
+        min_value=None,
+        max_value=None,
+        step=None,
+        format=None
+    ):
+        """Display a single-line numeric input widget.
+
+        Parameters
+        ----------
+        label : str or None
+            A short label explaining to the user what this input is for.
+        value : int/float or None
```

Comment:
it says `int/float` or `None` but we throw an error if they pass `None` I think

Reply:
That's something tricky because we could call the function with no value which would be None but if the user passes None we need to raise an exception